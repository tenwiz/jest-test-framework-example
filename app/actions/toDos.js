export const CREATE_USER = 'CREATE_USER'

export const createUser = username => ({
  type: CREATE_USER,
  username
})
