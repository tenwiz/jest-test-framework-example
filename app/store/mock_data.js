export const mock_todo = {
  username: 'Martian',
  task: 'Write test',
  color: 'red',
  deadline: 'today',
  complete: false,
  todoId: '123'
}
