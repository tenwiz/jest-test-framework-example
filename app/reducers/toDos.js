import {CREATE_USER} from "../actions/toDos";

const toDos = (state = {}, action) => {
  const { type, username } = action

  switch (type) {
    case CREATE_USER :
      return {
        ...state,
        [username]: {
          username,
          userToDos: []
        }
      }
    default :
      return state
  }
}

export default toDos
