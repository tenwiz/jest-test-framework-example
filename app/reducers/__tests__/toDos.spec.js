import configureStore from 'redux-mock-store'

import * as toDosActions from '../../actions/toDos'

import { mock_todo } from '../../store/mock_data'

import toDOs from '../toDos'

const middlewares = []
const mockStore = configureStore( middlewares )

const initialState = {}
const store = mockStore( initialState )

beforeEach( () => {
  store.clearActions();
} );

test( 'Select toDOs reducer', () => {
  const { username } = mock_todo
  const action = toDosActions.createUser( username )

  expect( toDOs( undefined, action ) ).toMatchSnapshot();
} );
