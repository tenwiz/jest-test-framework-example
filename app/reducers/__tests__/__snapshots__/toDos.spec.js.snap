// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select toDOs reducer 1`] = `
Object {
  "Martian": Object {
    "userToDos": Array [],
    "username": "Martian",
  },
}
`;
